code/game/objects/machinery/overwatch.dm

```diff
@@ -25,12 +30,37 @@ GLOBAL_LIST_EMPTY(active_cas_targets)
 	var/x_offset_s = 0
 	var/y_offset_s = 0
 	var/living_marines_sorting = FALSE
-	var/busy = FALSE //The overwatch computer is busy launching an OB/SB, lock controls
-	var/dead_hidden = FALSE //whether or not we show the dead marines in the squad.
-	var/z_hidden = 0 //which z level is ignored when showing marines.
-	var/datum/squad/current_squad = null //Squad being currently overseen
-	var/obj/selected_target //Selected target for bombarding
+	var/busy = FALSE ///The overwatch computer is busy launching an OB/SB, lock controls
+	var/dead_hidden = FALSE ///whether or not we show the dead marines in the squad.
+	var/z_hidden = 0 ///which z level is ignored when showing marines.
+	var/datum/squad/current_squad = null ///Squad being currently overseen
+	var/obj/selected_target ///Selected target for bombarding
+	var/current_order = NO_ORDER ///Selected order to give to marine
+	var/datum/action/innate/attack_order/send_attack_order ///datum used when sending an attack order
+	var/datum/action/innate/retreat_order/send_retreat_order ///datum used when sending a retreat order
+	var/datum/action/innate/defend_order/send_defend_order ///datum used when sending a defend order
```

Comment:
make this a list of orders.

Reply:
I've tried and it doesn't work really well with action etc.